packages/console/src/components/Steps/components/StepItem/Status.tsx

```diff
@@ -0,0 +1,38 @@
+import { conditional } from '@silverhand/essentials';
+import classNames from 'classnames';
+
+import Failed from '@/assets/images/failed.svg';
+import Success from '@/assets/images/success.svg';
+import { Ring } from '@/components/Spinner';
+
+import * as styles from './Status.module.scss';
+
+export type StepStatus = 'finished' | 'failed' | 'pending' | 'loading';
+
+type Props = {
+  step: number;
+  status?: StepStatus;
+  className?: string;
+};
+
+const StatusIconMap: Partial<Record<Required<Props>['status'], SvgComponent>> = {
```

Comment:
😂 why don't just `Partial<Record<StepStatus, SvgComponent>>`. also curious about why we don't just put all icons here

Reply:
Since I extracted the status into a component and defined a type for it but forgot to update it here.

Reply:
> why we don't just put all icons here

Only the `finished` and `failed` status need an icon.

The loading state uses a spin component and the pending uses a number.

<img width="88" alt="image" src="https://github.com/logto-io/logto/assets/10806653/c561119e-9588-4a75-866c-d09300ee3d98">

Reply:
that doesn't mean you need a map for these two `SvgComponent`s

Reply:
Yeah, removed.